kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -125,13 +126,23 @@ protected ConfigMap desired(KafkaProxy primary,
                     .addToLabels(standardLabels(primary))
                     .addNewOwnerReferenceLike(ResourcesUtil.newOwnerReferenceTo(primary)).endOwnerReference()
                 .endMetadata()
-                .withData(Map.of(CONFIG_YAML_KEY, generateProxyConfig(primary, context)))
+                .withData(generateDataMap(primary, context))
                 .build();
         // @formatter:on
     }
 
-    String generateProxyConfig(KafkaProxy primary,
-                               Context<KafkaProxy> context) {
+    @NonNull
+    private Map<String, String> generateDataMap(KafkaProxy primary, Context<KafkaProxy> context) {
+        try {
+            return Map.of(CONFIG_YAML_KEY, generateProxyConfig(primary, context));
+        }
+        catch (IllegalConfigurationException e) {
```

Comment:
Would we be better to use a io.javaoperatorsdk.operator.api.reconciler.dependent.Dependent#reconcilePrecondition so that the dependant produced if the Configuration would be invalid?